### PR TITLE
11593-mock-fss-api-for-addfile-and-get-batchstatus

### PR DESCRIPTION
**Pull request by Shirin14926**
code changes and scs mock missing data

### CODE REVIEW

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 1]
```diff
@@ -142,5 +148,43 @@ public IActionResult CommitBatch([FromRoute] string batchId, [FromBody] List<Bat
...
+                                                  [FromHeader(Name = "X-MIME-Type"), SwaggerSchema(Format = "MIME")] string contentType,
+                                                  [FromHeader(Name = "X-Content-Size"), SwaggerSchema(Format = ""), SwaggerParameter(Required = true)] long? xContentSize,
+                                                  [FromBody] FileRequest attributes)
+        {
+            if (!string.IsNullOrEmpty(batchId))
+            {
+                var response = fileShareService.CheckBatchWithZipFileExist(batchId, fileName, configuration["HOME"]);
+                if (response)
+                {
+                    return StatusCode((int)HttpStatusCode.Created);
```
Comment (barrie-cooper) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs:168]: Consider using `StatusCodes.Status201Created` (from `Microsoft.AspNetCore.Http`) over `(int)HttpStatusCode.Created`.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 2]
```diff
@@ -142,5 +148,43 @@ public IActionResult CommitBatch([FromRoute] string batchId, [FromBody] List<Bat
...
             return BadRequest(new { CorrelationId = GetCurrentCorrelationId(), Errors = ErrorsCommitBatch });
         }
+
+
+        [HttpPost]
+        [Route("batch/{batchId}/files/{fileName}")]
+        [Produces("application/json")]
+        [Consumes("application/json")]
+        public IActionResult AddFileToBatch([FromRoute, SwaggerSchema(Format = "GUID"), SwaggerParameter(Required = true)] string batchId,
+                                                  [FromRoute, SwaggerParameter(Required = true)] string fileName,
```
Comment (barrie-cooper) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs:158]: Tweak alignment please.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 3]
```diff
@@ -142,5 +148,43 @@ public IActionResult CommitBatch([FromRoute] string batchId, [FromBody] List<Bat
...
+                                                  [FromHeader(Name = "X-MIME-Type"), SwaggerSchema(Format = "MIME")] string contentType,
+                                                  [FromHeader(Name = "X-Content-Size"), SwaggerSchema(Format = ""), SwaggerParameter(Required = true)] long? xContentSize,
+                                                  [FromBody] FileRequest attributes)
+        {
+            if (!string.IsNullOrEmpty(batchId))
```
Comment (barrie-cooper) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs:163]: Also need to check at least `fileName`.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 4]
```diff
@@ -142,5 +148,43 @@ public IActionResult CommitBatch([FromRoute] string batchId, [FromBody] List<Bat
...
+                    return StatusCode((int)HttpStatusCode.Created);
+                }
+            }
+            return StatusCode((int)HttpStatusCode.InternalServerError, new { CorrelationId = GetCurrentCorrelationId(), Errors = ErrorsAddFileinBatch });
+        }
+
+        [HttpGet]
+        [Route("/batch/{batchId}/status")]
+        [Produces("application/json")]
+        public IActionResult GetBatchStatus([FromRoute][Required] string batchId)
```
Comment (barrie-cooper) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs:177]: Combine attributes i.e. `[FromRoute, Required]`.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Models/Request/FileRequest.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+using System.Diagnostics.CodeAnalysis;
+
+namespace UKHO.SalesCatalogueFileShareServicesMock.API.Models.Request
+{
+    public class FileRequest
+    {
+        public IEnumerable<FileAttribute> Attributes { get; set; }
+    }
+
+    public class FileAttribute
```
Comment (janvi13603) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Models/Request/FileRequest.cs:11]: Kindly create separate class file for FileAttribute

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Models/Response/BatchStatusResponse.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace UKHO.SalesCatalogueFileShareServicesMock.API.Models.Request
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Models/Response/BatchStatusResponse.cs]: Thanks for identify the typographical error.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 7]
```diff
@@ -142,5 +148,43 @@ public IActionResult CommitBatch([FromRoute] string batchId, [FromBody] List<Bat
...
+        {
+            if (!string.IsNullOrEmpty(batchId))
+            {
+                BatchStatusResponse batchStatusResponse = fileShareService.GetBatchStatus(batchId, configuration["HOME"]);
+                if (batchStatusResponse.Status == "Committed")
+                {
+                    return new OkObjectResult(batchStatusResponse);
+                }
+            }
+            return StatusCode((int)HttpStatusCode.Unauthorized);
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 8]
```diff
@@ -142,5 +148,43 @@ public IActionResult CommitBatch([FromRoute] string batchId, [FromBody] List<Bat
...
+        {
+            if (!string.IsNullOrEmpty(batchId))
+            {
+                var response = fileShareService.CheckBatchWithZipFileExist(batchId, fileName, configuration["HOME"]);
+                if (response)
+                {
+                    return StatusCode((int)HttpStatusCode.Created);
+                }
+            }
+            return StatusCode((int)HttpStatusCode.InternalServerError, new { CorrelationId = GetCurrentCorrelationId(), Errors = ErrorsAddFileinBatch });
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs (hunk 4 above)
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.
Thanks!

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs (hunk 1 above)
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs (hunk 3 above)
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Yes. Did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Models/Response/BatchStatusResponse.cs [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+{
+
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Models/Response/BatchStatusResponse.cs]: removed extra line

**Review by janvi13603 — APPROVED** [review]
Thanks for fixing review comments.

**Review by barrie-cooper — APPROVED** [review]
LGTM

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**